### PR TITLE
New post: Understanding streaming data use cases and applications - how to move real time analytics

### PR DESCRIPTION
**Pull request by cherbk**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [ ] The title of this pull request reflects the title of my content.
- [ ] I have reviewed the [Content Review Checklist](https://github.com/build-on-aws/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [ ] This pull request contains one piece of content (post or page).

Is this pull request linked to a Content Proposal or Idea Suggestion Issue? If so, please link to it below using the number or full link.

```text
Resolves #5

or

Content for https://github.com/build-on-aws/content/issues/5
```

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](/LICENSE).**

### CODE REVIEW

**Comment by lenadroid** [comment]
@abhirockzz could you please take a look for technical review? thanks

**Comment by lenadroid** [comment]
@franklinaws thanks so much for submitting your post! 

> Please ensure the file size is below 2000 Kb

Could you please reduce the size of the images to resolve validation errors? Thank you!

After you do, you should be able to see a nice preview link with what your post looks like on the website.

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+## Understanding the Stream store
+
+Let’s take a step back and talk about the data structure that supports streaming data architectures. This data structure is the commit log. A commit log works like a persistent queue composed of an ordered collection of records. Like a queue, a commit log can only be appended at the end, so older records are at the head and newer records are at the tail of the queue. The commit log is persistent in the sense that the data stored in it is not transient and can be analyzed any number of times. The data stored in a commit log can span seconds, minutes, days, months, or years!
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:78]: Not sure if this is accurate. I would consider re-thinking this. The ability to replay (and analyse) past data relies on persistence - it's not the other way around. These should be treated differently.  "data stored in a commit log" - this is good from a "mental model" point of view, but you need to mention the actual persistent store behind the commit log (disk, object store, etc.)

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+![FIGURE 7 Shows a line of kinds, similar to a queue](images/Figure7-streamstorequeue.jpg "Source: Adobe Stock")
+
+A commit log is a repository of all the mutations and events. A relational database is a subset of what is contained in the commit log. The values, mutations, and events in a commit log are captured as a snapshot of a point in time within tables in a relational database and presented using indexes to allow for querying and searching. Behind the scenes, the data you are querying databases against is a subset of the same data contained in the commit log. Going back to the fact that a commit log is really an ordered collection of data, a commit log is the ideal data structure for writing and storing streams of events, which is essentially what we call a stream store. Querying and reading through a stream store is different from a relational database and requires a different approach altogether.
+
+## Stream programming constructs
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:84]: Please add a sentence right before this, to help with a logical transition from "stream store" discussion into "Stream programming constructs"

Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:84]: BTW, ddi you mean "stream processing" (not programming?). Please consider replacing "programming" with "processing"

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+There are different programming constructs when working with streaming architectures.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:86]: remove the word "programming". and perhaps replace "constructs" with "concepts". will that make sense?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+* **Aggregations**: This is simply the summarization of data to query for particular values and allow you to perform operations like sum, average, min, max, etc.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:88]: I like this point. Do you think it makes sense to generalise it by stating "stateful computations" and aggregations being one of them?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+* **Causality and Relationships**: Stream processing also allows for detecting patterns to correlate or analyze the causality between events. This allows you to understand the relationships between one event and other events, enabling you to discern whether event A is tied to event B.
+
+![FIGURE 8 Shows the different stream programming constructs](images/Figure8-constructs.png)
+
+Leveraging these different programming constructs allows you to take advantage of what stream processing has to offer. These constructs can be used in isolation or in tandem to allow you to create simple or complex stream queries that effectively provide insight into the most up-to-date events and data emitted by your system. Ultimately, these constructs allow you to gain meaningful insights to make timely data-driven business decisions.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:94]: reconsider removing this line. these concepts are core to stream processing, not an add-on. and one would "implicitly" take advantage of them if adopting a stream processing architecture

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+## Stream processing considerations
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:96]: just like the previous suggestion. add a line before this heading to allow for a smooth logical transition

Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:96]: also, add a line after this as well (before jumping into the considerations)

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+### Transaction data into stream
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:98]: consider changing this heading - it's unclear

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+One consideration for organizations moving into real-time streaming would be transforming existing transaction data into a stream of events. Most organizations have transactional data residing in relational databases. The data in these databases is helpful in correlating with other streams of events and finding useful, actionable insights for your business. There are existing technologies that allow your transactional data stream to be captured using a technique called change data capture (CDC). This allows database systems to read at the log level using the commit log. For new systems, you should start writing streams of events directly into the stream store, thereby avoiding the otherwise two-step process of saving to a database and then generating streams through a CDC connector.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:100]: did you mean "transactional" data?

Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:100]: "This allows database systems to read at the log level using the commit log." - please change/reword this sentence.

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+![FIGURE 9 Transforming relation databases to real-time streaming](images/Figure9-cdc.png)
+
+### Avoid stream stores per project
+
+In the beginning, we talked about the challenges of building a silo-based architecture where your datastores are isolated. A similar scenario can arise when organizations build isolated stream stores. It’s ideal to have a common stream store across multiple systems, centralizing the stream store for all the different datasources to write into and having individual applications read and query from it. This simplifies access and the need to scale and maintain these isolated stream stores. Having a central data store also benefits data discovery across the organization.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:106]: consider changing to "In one of the previous section, we discussed the challenges...."

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+![FIGURE 12 Unidirectional flow](images/Figure13-unidirectionalflow.png)
+
+## Understanding build versus buy
+
+Last, but definitely not least, is understanding when it is best to build or buy. There are three major criteria to consider when deciding between building versus buying. First is competency. Is your team able to build and maintain an infrastructure? In the beginning, we talked about data analytics becoming a core function in the business. It's important for the organization to evaluate whether it makes business sense for your team to build and maintain this in-house. This is a decision not to be taken lightly, as it will take resources away from other activities. As for opting to buy and outsource your stream source to a vendor, assess their feature portfolio for completeness. Make sure you also look beyond what the stream store provides and inspect their stream processing framework, along with the tooling around it. Make sure they provide the features you need now and in the near future. It is not a trivial exercise to change implementation when you already have a central store in place and have our valuable insights.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:130]: change to "In the beginning, we discussed..."

Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:130]: "three major criteria" - please make sure to highlight each one of them individually (just like you did for the first one)

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+![FIGURE 12 Unidirectional flow](images/Figure14-buildvsbuy.jpeg "Source: Adobe Stock")
+
+## Conclusion
+
+In this blog, I talked about how data analytics have evolved from traditional data analytics, which deals with data in hours, days, months, or years, into real-time analytics, which deals with data in seconds, minutes, and hours. How siloed architecture needed to be bridged together. I looked at the current trends and how advancements in technology are ushering in real-time analytics to gain real-time insights that benefit the business. Understanding that real-time streaming use cases and applications are evolving our data analytics now and that there are important considerations when planning for your real-time streaming analytics platform is a great way to start on the right footing. This should equip you on your journey to start building out your own real-time analytics for your organization.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:136]: change "from traditional data analytics" to "from a traditional approach..."

Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:136]: the first sentence can be made much more concise

Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:136]: "How siloed architecture needed to be bridged together." - remove this or integrate this as part of another sentence.

**Review by abhirockzz — CHANGES_REQUESTED** [review]
This is a nice article! However, the promise of "how to move to real time analytics" is not reflected in the content. It does a great job of talking about the primitives, concepts and best practices, but it does not give prescriptive guidance on "how to actually move". 

A better way to explain would be by demonstrating how to shift from a batch architecture to real-time streaming with an example (even a simple one would have been fine). An actual step-by-step architecture and demo would be ideal. If that's too much, at least a theoretical discussion is warranted.

If the above cannot be done, consider changing the title and abstract to better suit the actual content.

**Comment by franklinaws** [comment]
> 

Hi Abhishek!  Thanks for reviewing and comments on the blogonaws article
Would you recommend a title change to any of this?

Understand the evolution from batched processing to real-time streaming
What to consider when moving from batch to realtime data stream processing
6 things to consider when moving from batch to realtime data stream processing

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 12]
```diff
@@ -0,0 +1,136 @@
+---
+title: Understanding streaming data use cases and applications - how to move real time analytics
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:2]: I made the changes, but would like to get your recommendation among these titles

Understand the evolution from batched processing to real-time streaming
What to consider when moving from batch to realtime data stream processing
6 things to consider when moving from batch to realtime data stream processing

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+description: Learn how we evolved into using real-time streaming analytics.  We'll take a step back to deep-dive on streaming concepts and use cases, understand how to move your organization to leverage real-time streaming.
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:3]: sorry,  where can i see the double space?

[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:3]: reworded the description

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+tags:
+  - analytics
+  - big-data
+  - real-time
+  - traditional-analytics
+  - batch-processing
+  - perishable-insights
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:10]: Removed

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+  - commit-log
+  - stream-store
+  - stream-ingestion
+authorGithubAlias: aguinalf
+authorName: Franklin Aguinaldo
+date: 2023-06-13
+---
+
+A lot of things have changed in the past few years in how we handle data, from introducing different ways to store it using data warehousing, Hadoop, and NoSQL databases to offering remote storage and faster compute through cloud computing. Advancements in networking and computing have pushed more processing to happen at the edge with edge computing and the Internet of Things. Now we are seeing a diverse set of data sources sending data more often and at higher quality.
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:19]: Rephrased

Advancements in networking and computing have pushed more processing to happen at the edge with Internet of Things and edge computing. Now we are seeing a diverse set of data sources sending data more often and at higher quality.

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+
+![FIGURE 2 showing a graphic of hard drives, to show how data we capture and store has grown over the decades](images/Figure2-hard-drive.jpeg "Source: Adobe Stock")
+
+In 2021, Gartners released an article on the Top 10 Data and Analytics Trends for 2021. In this article, Gartner mentioned two trends where they cite data and analytics as core business functions. They cited how business leaders are beginning to understand the importance of using data and analytics to accelerate digital business initiatives, shifting from a secondary focus to a core function for the business. In that same article, Gartner talks about how more data analytics technologies are starting to live outside of the traditional data center and cloud environments, closer to the physical assets. This means doing data analytics at the edge to reduce latency for data-centric solutions and enable more real-time value. The renewed focus on data analytics and the need for more real-time processing have evolved analytics to put more emphasis on processing in real-time.
+
+## Perishable insights
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:29]: Removed the heading.

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+
+In 2022, Forrester released a report on "perishable insights". The report talks about how the value of data diminishes and perishes over time. Traditional analytics is designed to process data that is stored over a period of days. months and years. But for real-time systems, data is time-sensitive and is designed to be processed in a matter of seconds, minutes, or sometimes fractions of a second.
+
+![FIGURE 3 Shows Forrester's graphic on perishable insights](images/Figure3-perishableinsights-forrester.jpg)
+
+## Siloed-based architecture
+
+For most systems that start early on, they are designed using a siloed-based architecture.  Some start from small
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:37]: Done

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+
+![FIGURE 5 batch vs realtime](images/Figure5-batch-vs-realtime.png)
+
+There are certain processes where batch processing works well. Some examples are processing hourly logs, weekly or monthly bills, daily website clickstreams, or daily fraud reports. Again, these are data where they fall in traditional analytics, with the age of the data in the hours, days, months, or years.
+
+Some of the systems now are evolving into real-time systems that are pushing large volumes of data, requiring continuous processing and analysis of live data, allowing enterprises to gain insights in real time from varied sources like devices and systems. The issue with processing in batches is that you are getting insights from data that is not necessarily up-to-date. Real-time processing, as opposed to traditional analytics, processes and analyzes these events and data as they happen.
+
+## Real-time data streaming 
+
+Real-time data streaming is the collection and ingestion of data from various sources and processing it in real time to gain meaning and insight.
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:58]: Its similar, let me pick one.

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+
+### Components of Real-time Data Streaming
+
+![FIGURE 6 Real-time streaming components](images/Figure6-realtime-streaming.jpg)
+
+Above are the components of a real-time data streaming pipeline. The sources are devices or applications producing high volumes of continuous data at high velocity, like mobile devices, web applications, application logs, IoT sensors, smart devices, and gaming applications. The stream ingestion component provides the solution to capture continuous data from these sources and store it in scalable stream storage. The stream processing component allows you to transform and deliver data to a destination for further analysis or long-term storage. In each of the components of the processing, from the ingestion to the destination, the components need to be able to scale as the data volume increases.
+
+Below are some of the common use cases for streaming
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:66]: Thats a good catch.  I chose to simplify tackling the components into that paragraph.  Let me remove the components heading, add a use cases heading and rearrange the paragraphs a bit

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+
+## Real-time data streaming 
+
+Real-time data streaming is the collection and ingestion of data from various sources and processing it in real time to gain meaning and insight.
+
+### Components of Real-time Data Streaming
+
+![FIGURE 6 Real-time streaming components](images/Figure6-realtime-streaming.jpg)
+
+Above are the components of a real-time data streaming pipeline. The sources are devices or applications producing high volumes of continuous data at high velocity, like mobile devices, web applications, application logs, IoT sensors, smart devices, and gaming applications. The stream ingestion component provides the solution to capture continuous data from these sources and store it in scalable stream storage. The stream processing component allows you to transform and deliver data to a destination for further analysis or long-term storage. In each of the components of the processing, from the ingestion to the destination, the components need to be able to scale as the data volume increases.
```
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:64]: Thank you.  Don't mind if i do. :)

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 8 above)
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:100]: > did you mean "transactional" data?

Yes. changed.

Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:100]: > 

That sentence was not really helping this paragraph.  What i wanted to say is 
"This allows you to bring your transactional data into your streaming platform"

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 9 above)
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:106]: Changed

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 10 above)
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:130]: > change to "In the beginning, we discussed..."
Changed.

Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:130]: > "three major criteria" - please make sure to highlight each one of them individually (just like you did for the first one)

Good advice,  I modified it,

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 11 above)
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:136]: > the first sentence can be made much more concise

changed into
"In this blog, I talked about how data analytics have evolved from traditional data approach dealing with data in hours, days, months, or years, into data in seconds and minutes and hours through real-time analytics."

Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:136]: Incorporated into

"I looked at the current trends and how advancements in technology are ushering in real-time analytics to gain real-time insights that benefit the business and help break down data silos."

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md]: Thanks. I would actually go with this title "Six Things to Consider When Moving From Batch to Real-time Data Stream Processing" (replace six with a number you think that makes sense in-line with the content).

description looks good. if possible, try to make the description a bit more concise if possible (think "one-liner").

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md]: in the description - "analytics.  We'll"

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+
+## Types of data processing
+
+Traditional analytics involved two major phases: data getting stored and then analyzed. Data can be stored in a lot of different ways and formats, like files or in databases. One of the ways this data is processed is through batch processing, where stored data is processed and analyzed in batches. Batch processing is efficient at processing large amounts of data and is suitable when you don’t need fast results.
+
+![FIGURE 5 batch vs realtime](images/Figure5-batch-vs-realtime.png)
+
+There are certain processes where batch processing works well. Some examples are processing hourly logs, weekly or monthly bills, daily website clickstreams, or daily fraud reports. Again, these are data where they fall in traditional analytics, with the age of the data in the hours, days, months, or years.
```
[the remark this answers is not in the document]
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md]: hmmm. I would consider re-writing it as follows if possible "Batch processing excels at handling tasks like processing hourly logs, weekly or monthly bills, daily website clickstreams, or daily fraud reports. These involve data typically used in traditional analytics, where the age of the data spans from hours to days, months, or even years."

what do you think?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 1 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:74]: @franklinaws did you get a chance to look into this?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 11 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md]: I would consider changing to something like "In this blog, I discuss the evolution of data analytics from traditional methods, which handle data spanning hours, days, months, or years, to real-time analytics that process data within seconds, minutes, and hours."

Let me know what you think

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 22]
```diff
@@ -0,0 +1,132 @@
...
+
+![FIGURE 12 Unidirectional flow](images/Figure13-unidirectionalflow.png)
+
+## Understanding build versus buy
+
+Last, but definitely not least, is understanding when it is best to build or buy. There are three major criteria to consider when deciding between building versus buying. First is competency. Is your team able to build and maintain an infrastructure? In the beginning, we talked about data analytics becoming a core function in the business. It's important for the organization to evaluate whether it makes business sense for your team to build and maintain this in-house. This is a decision not to be taken lightly, as it will take resources away from other activities. Second is considering to buy and outsource your stream source to a vendor. Assess their feature portfolio for completeness. Make sure you also look beyond what the stream store provides and inspect their stream processing framework, along with the tooling around it. Third, make sure they provide the features you need now and in the near future. It is not a trivial exercise to change implementation when you already have a central store in place and have our valuable insights.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:126]: "Second is considering to buy.... " Can you update this to "Secondly, consider purchasing and outsourcing your stream source to a vendor." ?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+
+![FIGURE 12 Unidirectional flow](images/Figure14-buildvsbuy.jpeg "Source: Adobe Stock")
+
+## Conclusion
+
+In this blog, I talked about how data analytics have evolved from traditional data approach dealing with data in hours, days, months, or years, into data in seconds and minutes and hours through real-time analytics. I looked at the current trends and how advancements in technology are ushering in real-time analytics to gain real-time insights that benefit the business and help break down data silos. Understanding that real-time streaming use cases and applications are evolving our data analytics now and that there are important considerations when planning for your real-time streaming analytics platform is a great way to start on the right footing. This should equip you on your journey to start building out your own real-time analytics for your organization.
```
Comment (abhirockzz) [inline posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:132]: I would consider changing to something like "In this blog, I discuss the evolution of data analytics from traditional methods, which handle data spanning hours, days, months, or years, to real-time analytics that process data within seconds, minutes, and hours."

Let me know what you think

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 7 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:94]: did you get a chance to take a look at this?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 6 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:92]: did you get a chance to take a look at this?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 5 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:90]: did you get a chance to take a look at this?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 4 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:84]: did you get a chance to take a look at this?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 2 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:80]: did you get a chance to take a look at this?

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 1 above)
Comment (abhirockzz) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:74]: did you get a chance to take a look at this?

**Review by abhirockzz — CHANGES_REQUESTED** [review]
Thanks for the updates! 

As far as the title is concerned, I would actually go with this title "Six Things to Consider When Moving From Batch to Real-time Data Stream Processing" (replace six with a number you think that makes sense in-line with the content).

The description looks good. if possible, try to make it a bit more concise if possible (think "one-liner").

Please take a look at some of pending review items you might have missed and let me know what you think. Once that's done, we should be good to go.

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md]: I made the title change to what you recommended.  I also modified the description into "Learn how we evolved from batch into using real-time streaming analytics and understand what to consider when moving your organization to real-time streaming."

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md]: I rewrote the description.  It should be just one sentence

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 22 above)
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:126]: Done

File: posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md (hunk 23 above)
Comment (franklinaws) [reply posts/understanding-streaming-data-uses-cases-and-applications-how-to-move-to-real-time-analytics/index.md:132]: Thanks! Done.

**Review by riferrei — APPROVED** [review]
Content ready

**Comment by cobusbernard** [comment]
Thanks for the great post @franklinaws, we couldn't edit this PR, so we create a new one with some small fixes before we published - #531 , article is now published.

**Pull request closed without merging**